lib/src/common/repos_service.dart

```diff
@@ -309,16 +329,17 @@ class RepositoriesService extends Service {
     return false;
   }
 
+  /// https://docs.github.com/en/rest/collaborators/collaborators#add-a-repository-collaborator
   Future<bool> addCollaborator(RepositorySlug slug, String user) async {
     ArgumentError.checkNotNull(slug);
     ArgumentError.checkNotNull(user);
     return github
         .request(
           'PUT',
           '/repos/${slug.fullName}/collaborators/$user',
-          statusCode: StatusCodes.NO_CONTENT,
+          statusCode: StatusCodes.CREATED,
         )
-        .then((response) => response.statusCode == StatusCodes.NO_CONTENT);
+        .then((response) => response.statusCode == StatusCodes.CREATED);
```

Comment:
Hmm according to the docs here https://docs.github.com/en/rest/collaborators/collaborators#add-a-repository-collaborator--status-codes
It may return either 201 CREATED  or 204 NO_CONTENT depending on

201 | Response when a new invitation is created
-- | --
204 | Response when:an existing collaborator is added as a collaboratoran organization member is added as an individual collaboratoran existing team member (whose team is also a repository collaborator) is added as an individual collaborator

So I think the response checking might need to check both?
```suggestion
        .then((response) => response.statusCode == StatusCodes.CREATED || response.statusCode == StatusCodes. NO_CONTENT);
```